crates/types/src/retries.rs

```diff
@@ -124,25 +125,25 @@ impl Default for RetryPolicy {
 }
 
 impl RetryPolicy {
-    pub fn fixed_delay(interval: Duration, max_attempts: usize) -> Self {
+    pub fn fixed_delay(interval: Duration, max_attempts: Option<usize>) -> Self {
         Self::FixedDelay {
             interval: interval.into(),
-            max_attempts,
+            max_attempts: max_attempts.map(|m| NonZeroUsize::new(m).expect("non-zero")),
         }
     }
 
     pub fn exponential(
         initial_interval: Duration,
         factor: f32,
-        max_attempts: usize,
+        max_attempts: Option<usize>,
```

Comment:
Same question here.

Reply:
Just to avoid refactoring the high number of call sites at this PR.